viewing_party/party.py

```diff
@@ -1,23 +1,125 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    else:
+        movie = {}
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    movie = next((item for item in user_data["watchlist"] if item["title"] == movie_title), None)
+    if movie:
+        add_to_watched(user_data, movie)
+        user_data["watchlist"].remove(movie)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_total = 0
+    number_of_ratings = len(user_data["watched"])
+    for i in range(number_of_ratings):

```

Comment:
I saw a few places where `for i in range(...)` followed by a reference to a dictionary at some key at an index at another key. That totally works and is safe, however, you could simplify that by (in this case, but also in every case I saw that followed this pattern) doing something like:
```python
    for movie in user_data["watched"]:
        total_movie_ratings += movie["rating"]
```
which is a bit easier to read.